packages/webapp/src/components/PopupFilter/PureCollapsibleSearch/index.jsx

```diff
@@ -0,0 +1,136 @@
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState, useRef, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+import { ReactComponent as SearchIcon } from '../../../assets/images/search.svg';
+import Input from '../../Form/Input';
+import TextButton from '../../Form/Button/TextButton';
+import { Modal } from '../../Modals';
+
+export default function PureCollapsibleSearch({
+  value,
+  isSearchActive,
+  onChange,
+  placeholderText,
+  className,
+  containerRef,
+}) {
+  const searchRef = useRef(null);
+  const [modalStyle, setModalStyle] = useState({});
+
+  const [searchOverlayOpen, setSearchOverlayOpen] = useState(false);
+  const onSearchOpen = () => setSearchOverlayOpen(true);
+  const onSearchClose = () => setSearchOverlayOpen(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSearchClose();
+  };
+
+  // Position modal searchbar correctly
+  useLayoutEffect(() => {
+    let resizeObserver;
+
+    const updateModalStyle = (ref) => {
+      if (ref) {
+        const rect = ref.getBoundingClientRect();
+
+        setModalStyle({
+          position: 'absolute',
+          top: `${rect.top}px`,
+          left: ref === searchRef.current ? undefined : `${rect.left}px`,
+          width: ref === searchRef.current ? '95vw' : `${rect.width}px`,
+          margin: 0,
+          maxWidth: '100vw',
+        });
+      }
+    };
+
+    if (searchOverlayOpen) {
+      updateModalStyle(containerRef?.current || searchRef?.current);
+
+      // If containerRef is provided, observe  it for resize events
+      if (containerRef?.current) {
+        resizeObserver = new ResizeObserver(() => {
+          updateModalStyle(containerRef.current);
+        });
+        resizeObserver.observe(containerRef.current);
+      }
+    }
+
+    return () => {
+      if (resizeObserver) {
+        resizeObserver.disconnect();
+      }
+    };
+  }, [searchOverlayOpen]);
+
+  return (
+    <div className={clsx(styles.container, className)} ref={searchRef}>
+      <Input
+        className={styles.searchBar}
+        isSearchBar
+        value={value}
+        onChange={onChange}
+        placeholder={placeholderText}
+      />
+
+      {searchOverlayOpen && (
+        <Modal dismissModal={onSearchClose} style={modalStyle}>
+          <form // allows closing modal with enter key (= submit)
+            onSubmit={handleSubmit}
+            className={styles.modalContent}
+          >
+            <Input
+              className={styles.modalSearchbar}
+              isSearchBar
+              value={value}
+              onChange={onChange}
+              placeholder={placeholderText}
+            />
+          </form>
+        </Modal>
+      )}
+
+      <TextButton className={clsx(styles.searchButton, isSearchActive && styles.active)}>
+        <div className={styles.circleContainer}>
```

Comment:
@SayakaOno made me aware of this the other day, we shouldn't place a div inside a TextButton because it's a <button> tag, which doesn't allow nesting block elements inside it. Since the circle is positioned absolutely anyway, you can just remove it from the TextButton and place it outside. You'll need a container for both with `position: relative` for it to work properly.